From d9378defa9f4896e4685ee88af1a97126826369f Mon Sep 17 00:00:00 2001
From: Nanley Chery <nanley.g.chery@intel.com>
Date: Thu, 17 Feb 2022 10:26:34 -0800
Subject: [PATCH 01/25] anv: Don't disable HiZ/MCS in anv_BindImageMemory2

When an image is bound to a BO lacking implicit CCS, we disable any
compression it may have had. This is unnecessary in the cases where the
compression type doesn't depend on the BO having implicit CCS support.
Avoid this disabling for ISL_AUX_USAGE_MCS and ISL_AUX_USAGE_HIZ.

Reviewed-by: Lionel Landwerlin <lionel.g.landwerlin@intel.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/15068>
(cherry picked from commit ffbde42b9374c1b70b9caefa981e3c09bea0e7b1)
---
 src/intel/vulkan/anv_image.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/intel/vulkan/anv_image.c b/src/intel/vulkan/anv_image.c
index c844f292a8b..2efa792b12c 100644
--- a/src/intel/vulkan/anv_image.c
+++ b/src/intel/vulkan/anv_image.c
@@ -1834,6 +1834,9 @@ VkResult anv_BindImageMemory2(
          const struct anv_bo *bo =
             image->bindings[binding].address.bo;
 
+         if (!isl_aux_usage_has_ccs(image->planes[p].aux_usage))
+            continue;
+
          if (bo && !bo->has_implicit_ccs &&
              device->physical->has_implicit_ccs)
             image->planes[p].aux_usage = ISL_AUX_USAGE_NONE;
-- 
2.34.1

